### PR TITLE
[tests] Use a constant-length timestamp redaction for test determinism

### PR DESCRIPTION
**Pull request by smklein**
Changes our "timestamp redaction" logic to ignore the original length of the input timestamp.

Timestamps are not always constant-width -- depending on the amount of trailing zeros, they can have variable stringified lengths. Unfortunately, this variability is at-odds with our goal of deterministic test output.

In this PR, we favor the deterministic output, to avoid test flaking: Variable space output logic is removed in favor of a stable redaction marker. Additionally, tests with different timestamp lengths are added to act as a regression against this particular flake.

Fixes https://github.com/oxidecomputer/omicron/issues/6509

### CODE REVIEW

File: test-utils/src/dev/test_cmds.rs [hunk 1]
```diff
@@ -329,4 +322,19 @@ mod tests {
              path3: <VARIABLE_REDACTED>"
         );
     }
+
+    #[test]
+    fn test_redact_timestamps() {
+        let times = [DateTime::<Utc>::from_timestamp_nanos(0), Utc::now()];
```
Comment (iliana) [inline test-utils/src/dev/test_cmds.rs:328]: Somewhat ironically I worry about `Utc::now` triggering flakes later. Could we loop through a bunch of values for `from_timestamp_nanos` to test the varying subsecond lengths?

File: test-utils/src/dev/test_cmds.rs [hunk 2]
```diff
@@ -155,32 +155,24 @@ pub fn redact_variable(input: &str) -> String {
...
+    // Replace timestamps that have a sub-second portion.
+    // Example: 1970-01-01T00:00:00.000001Z
+    //
+    // Note that depending on the amount of trailing zeros,
+    // this value can have different widths. However, "<REDACTED_TIMESTAMP>"
+    // has a deterministic width, so that's used instead.
+    let s = regex::Regex::new(r"\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\.\d+Z")
+        .unwrap()
+        .replace_all(&s, "<REDACTED_TIMESTAMP>")
+        .to_string();
```
Comment (iliana) [inline test-utils/src/dev/test_cmds.rs:175]: GitHub won't let me write a suggestion here but I suspect combining these into a single regex would work? e.g. `r"\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}(\.\d+)?Z"`

**Review by iliana — APPROVED** [review]

File: test-utils/src/dev/test_cmds.rs (hunk 1 above)
Comment (smklein) [reply test-utils/src/dev/test_cmds.rs]: Sure thing, updated with with a bunch of stable values.

File: test-utils/src/dev/test_cmds.rs (hunk 2 above)
Comment (smklein) [reply test-utils/src/dev/test_cmds.rs]: That seems valid -- updated!

**Comment by davepacheco** [comment]
Thanks for this fix!

**Pull request merged**